### DISCUSSION TITLE
Sprite2D - needs size x, y

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9911] [Editor]
PROBLEM: If you want to match the sprite size to the collision shape size in code it is math and bug heavy.

EXAMPLE: A simple sprite a user can click and move: Area2D a CollisionShape2D (rect) and a Sprite2D

1. The collision rectangle uses size x and y (essentially width and height).

2. The sprite uses scale. So now you're dealing with floats and ints and division. Disaster and a regular source of bugs.

PROPOSAL: I'd like a size property on sprite2D, x and y, width and height.

**Comment by Mickeon** [comment]
> So now you're dealing with floats and ints and division

Just for clarity, there are no integers, it's all floats.

You can avoid this whole process by using the same `scale` as the Sprite2D on the CollisionShape2D (I do believe works? I've done this myself in the past.)

**Reply by shak2** [reply]
If they can both have scale, then they should both have size!

**Reply by AThousandShips** [reply]
Why, how are they related? The sprite gets it's size from the texture, what would setting the size do? Would it clip it?

**Comment by AThousandShips** [comment]
This feels like something that doesn't fit most workflows for designing 2D physics to match the visuals, it's usually (in my experience) not a simple numbers game but a visual process, rarely does the shape "just fit" the sprite bounds, and you usually want margins, etc.

I doubt this specific use case would be used much
